lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -467,7 +484,13 @@ pub extern "C" fn wasm_engine_new_with_config(
                         // There are currently no uses of the object-file engine + compiler from the C API.
                         // So we run in headless mode.
                         if #[cfg(feature = "object-file")] {
-                            Arc::new(ObjectFile::headless().engine())
+                            let mut builder = ObjectFile::headless();
+
+                            if let Some(target) = config.target {
+                                builder = builder.target(target.inner);
```

Comment:
Does it make sense to set the target for a headless-only engine?

It's fine for symmetry I suppose but I don't think this can ever do anything

Reply:
Yeah I was wondering if that was making sense. Using the target makes sense _only_ when compiling, but we never know, maybe one day, someone, somewhere… :-p. Let's keep it for “the symmetry” as you said, how does it sound?